src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,353 @@
+        testfile = os.path.join("/", file_name)
+        ior_cmd.test_file.update(testfile)
+
+        manager = get_job_manager(
+            test=self, class_name="Mpirun", job=ior_cmd, subprocess=self.subprocess,
+            mpi_type="mpich", timeout=timeout)
+        manager.assign_hosts(
+            self.hostlist_clients, self.workdir, self.hostfile_clients_slots)
+
+        # Run the command.
+        try:
+            self.log.info("--- IOR command %d start ---", job_num)
+            ior_output = manager.run()
+            results[job_num] = [True]
+            # For debugging.
+            results[job_num].extend(IorCommand.get_ior_metrics(ior_output))
+            # Command worked, but append the error message if any.
+            results[job_num].append(ior_output.stderr_text)
+            self.log.info("--- IOR command %d end ---", job_num)
+        except CommandFailure as error:
+            self.log.info("--- IOR command %d failed ---", job_num)
+            results[job_num] = [False, "IOR failed: {}".format(error)]
+
+    @staticmethod
+    def check_container_health(container, expected_health):
+        """Check container property's Health field by calling daos container get-prop.
+
+        Args:
+            container (TestContainer): Container to call get-prop.
+            expected_health (str): Expected value in the Health field.
+
+        Returns:
+            bool: True if expected_health matches the one obtained from get-prop.
+
+        """
+        output = container.get_prop(properties=["status"])
+        actual_health = output["response"][0]["value"]
+
+        return actual_health == expected_health
+
+    def verify_ior_worked(self, ior_results, job_num, errors):
+        """Verify that the IOR worked.
+
+        Args:
+            ior_results (dict): Dictionary that contains the IOR run results.
+            job_num (int): Job number used for the IOR run.
+            errors (list): Error list used in the test.
+        """
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append("Error found in second IOR run! {}".format(ior_error))
+
+    def update_network_interface(self, interface, state, host, errors=None):
+        """Bring back or tear down the given network interface.
+
+        Args:
+            interface (str): Interface name such as ib0.
+            state (str): "up" or "down".
+            host (str): Host to update the interface.
+            errors (list): List to store the error message, if the command fails.
+                Defaults to None.
+        """
+        command = "sudo ip link set {} {}".format(interface, state)
+        results = run_pcmd(hosts=[host], command=command)
+        self.log.info("%s output = %s", command, results)
+        if errors is not None and results[0]["exit_status"] != 0:
+            errors.append("{} didn't return 0!".format(command))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string